ibis/expr/operations.py

```diff
@@ -720,6 +720,17 @@ class Quantile(Reduction):
     output_type = rules.type_of_arg(1)
```

Comment:
So, I think this needs to be:

```python
def output_type(self):
    first_arg = self.args[0]
    first_arg_type = first_arg.type()
    if isinstance(first_arg_type, dt.Integer):
        result_type = dt.double.scalar_type()
    else:
        result_type = first_arg_type
    return result_type.scalar_type()
```
because the result will be a double if the input is an integer and it will take on the type of the first argument otherwise.